src/raster/processing/dem/options.rs

```diff
@@ -0,0 +1,172 @@
+use std::fmt::{Display, Formatter};
+use std::marker::PhantomData;
+use std::ptr;
+use std::ptr::NonNull;
+
+use gdal_sys::{
+    GDALDEMProcessingOptions, GDALDEMProcessingOptionsFree, GDALDEMProcessingOptionsNew,
+};
+
+use crate::cpl::CslStringList;
+use crate::errors;
+use crate::utils::_last_null_pointer_err;
+
+/// Payload for [`GDALDEMProcessing`]. Intended for internal use only.
+pub struct GdalDEMProcessingOptions<'opts>(
+    NonNull<GDALDEMProcessingOptions>,
+    PhantomData<&'opts CslStringList>,
+);
+
+impl<'opts> GdalDEMProcessingOptions<'opts> {
+    pub fn new(opts: &'opts CslStringList) -> errors::Result<Self> {
+        let popts = unsafe { GDALDEMProcessingOptionsNew(opts.as_ptr(), ptr::null_mut()) };
+        if popts.is_null() {
+            return Err(_last_null_pointer_err("GDALDEMProcessingOptionsNew"));
+        }
+        Ok(Self(unsafe { NonNull::new_unchecked(popts) }, PhantomData))
```

Comment:
Let's use `NonNull::new` and match on that, it's not uglier than checking `popts`. We could even use https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html, but only after a MSRV bump (we're on 1.58? what happened to "`stable` + 3" or whatever policy we had?).

Reply:
https://github.com/metasim/gdal/pull/1

Reply:
Nice!